packages/deployer/src/deployUtils.ts

```diff
@@ -0,0 +1,153 @@
+      'function decimals() view returns (uint256)'
+    ], ethers.provider)
+  // verify token address is a valid token...
+  const symbol: string = await token.symbol().catch((e: any) => null)
+  const decimals: number = await token.decimals().catch((e: any) => null)
+  if (symbol == null || decimals == null) {
+    throw new Error(`invalid token: ${address} (Symbol: ${symbol} Decimals: ${decimals})`)
+  }
+  const divisor = Math.pow(10, decimals)
+
+  return {
+    address,
+    symbol,
+    decimals,
+    balanceOf: async (addr: string) => token.balanceOf(addr).then((v: any) => v.div(divisor))
+  }
+}
+
+
+/** helper: set a field on a contract only if it was changed.
+ * the "deploy" mechanism has the property of "re-deploy only on change". this method replicate the logic for calling a setter.
+ * @param deployments - hardhat deploy extension
+ * @param contract - a contract deployed earlier using "deploy"
+ * @param getFunc - a getter (with no params) to read current value
+ * @param setFunc - a setter function (accepting a single value) to set the new value
+ * @param val - the value we want the field to have
+ * @param deployer
+ */
+export async function setField (deployments: DeploymentsExtension, contract: string, getFunc: string, setFunc: string, val: string, deployer: string): Promise<void> {
+  const options: TxOptions = {
+    from: deployer,
+    log: true
+  }
+  const currentVal = await deployments.read(contract, options, getFunc)
+  if (currentVal !== val) {
+    console.log('calling', `${contract}.${setFunc}( ${val} )`)
+    await deployments.execute(contract, options, setFunc, val)
+  }
+}
+
+async function getStakingInfo (hre: HardhatRuntimeEnvironment, env: Environment): Promise<{ stakingTokenAddress: string, stakingTokenValue: string }> {
+  // @ts-ignore
+  let stakingTokenAddress = Object.keys(env.deploymentConfiguration.minimumStakePerToken ?? {})[0]
+  // @ts-ignore
+  const stakingTokenValue = env.deploymentConfiguration.minimumStakePerToken[stakingTokenAddress]
+
+  if (stakingTokenAddress === 'test') {
+    stakingTokenAddress = await hre.deployments.get('WrappedEthToken').then(res => res.address)
+  }
+  return { stakingTokenAddress, stakingTokenValue }
+}
+
+export async function printRelayInfo (hre: HardhatRuntimeEnvironment): Promise<void> {
+  const { getChainId } = hre
+  console.log('Connected to URL: ', (hre.network.config as HttpNetworkConfig).url)
+  const accounts = await ethers.provider.listAccounts()
+  const deployer = accounts[0]
+
+  const chainId = parseInt(await getChainId())
+  const env: Environment = getMergedEnvironment(chainId, deployer)
+
+  const { stakingTokenAddress, stakingTokenValue } = await getStakingInfo(hre, env)
+
+  const hub = await hre.deployments.get('RelayHub')
+  const network = hre.network.config as HttpNetworkConfig
+  console.log(chalk.white('Relayer config:'))
+  console.log(chalk.grey(JSON.stringify({
+    baseRelayFee: 0,
+    pctRelayFee: 70,
```

Comment:
What is going on here? Who is this output aimed for? Why does it have hard-coded values?